### ISSUE TITLE
Add named parameters in mapping types with solidity 0.8.18

### ISSUE THREAD

**Issue by cupOJoseph** [transmissions11/solmate#360]
I'm happy to make the changes if maintainers want to assign me, but it will require updating the contracts to 0.8.18, which may not be ideal at the moment.

Example: ERC20 could be changed to 

```javascript
    /*//////////////////////////////////////////////////////////////
                              ERC20 STORAGE
    //////////////////////////////////////////////////////////////*/

    uint256 public totalSupply;

    mapping(address user => uint256 balance) public balanceOf;

    mapping(address user => mapping(address spender => uint256 allowance)) public allowance;
```

**Comment by transmissions11** [comment]
> I'm happy to make the changes if maintainers want to assign me, but it will require updating the contracts to 0.8.18, which may not be ideal at the moment.

yeahh my thoughts as well... we'll have to make a decision about breaking backwards compatibility soon cuz it seems like 0.9 is a ways away but idt this is the time yet

**Comment by cupOJoseph** [comment]
I agree there are benefits for everything to just be <=0.8.0 for now. I'll check in again on this closer to 0.9
